### PR TITLE
Feature/207 set up state for sockets

### PR DESCRIPTION
**Pull request by bryansamuel1**
**Work Completed**

1. Add namespaces to all files that use the store, replacing any hardcoded namespaces with an enum
2. Split the store modules in two. The app module contains all the modules that have state that is not shared via the sockets. The socket module contains all the modules that have state that is shared via the sockets. 
3. Move data that is stored in the @Component data object into the class so it is a class property. This makes it consistent throughout the app
4. Rename some components so they following the naming convention of the app. Components that are only used once are now prepended with The (TheIndex, TheRoom) etc.
5. Cleaned up some of the formatting throughout the app so indentation is consistent and spacing between classes is consistent

### CODE REVIEW

File: src/components/canvas-tools/ToolContainer.vue [hunk 1]
```diff
@@ -27,7 +27,7 @@
               <v-icon dense :color="iconColour">{{icon}}</v-icon>
             </v-btn>
           </template>
-          <span>{{ $t(`tool.container.${toolName}`) }}</span>
+          <span>{{ $t(`tool.container.${toolname}`) }}</span>
```
Comment (Eirmas) [inline src/components/canvas-tools/ToolContainer.vue:30]: dont we want to use cascalCase here?

**Comment by bryansamuel1** [comment]
> Good and efficient work :)
> We could also use namespaces inside the modules for the mutations and actions and all that

I could do that now it shouldn't take me long

**Comment by bryansamuel1** [comment]
> > Good and efficient work :)
> > We could also use namespaces inside the modules for the mutations and actions and all that
> 
> I could do that now it shouldn't take me long

I'll do it in another PR. This one is large enough

File: src/components/canvas-tools/ToolContainer.vue (hunk 1 above)
Comment (bryansamuel1) [reply src/components/canvas-tools/ToolContainer.vue]: Fixed. It is now camelCase

**Comment by bryansamuel1** [comment]
With my latest commit I have also done the following

1) Changed the disabled colour for the entity panel to the same colour as the nav
2) Put a button saying "Login" instead of the menu dropdown. If they are logged in then the menu drop down will show as normal
3) Tidied up some of the code from Nath PR just to tie in with the rest of the clean up and refactoring that I have done in this PR
4) Fix PR comment from Eirmas
5) Made the provider PascalCase with spaces instead of lowercase (asked by @NikoGrano previously)

**Comment by bryansamuel1** [comment]
Here is the new Login button for the nav
image

**Comment by NikoGrano** [comment]
Please check always preview!

**Comment by NikoGrano** [comment]
There is one thing missing from #187.

When user has logged in, the `GET` parameter `code` should be removed from url to prevent users copying the url and giving it to somebody else as "please join room via this link".

In that case they would also give their session to that user....

**Comment by bryansamuel1** [comment]
> Please check always preview!

You wha?

**Comment by bryansamuel1** [comment]
> There is one thing missing from #187.
> 
> When user has logged in, the `GET` parameter `code` should be removed from url to prevent users copying the url and giving it to somebody else as "please join room via this link".
> 
> In that case they would also give their session to that user....

agreed. I will do that as part of this PR

**Comment by NikoGrano** [comment]
> > Please check always preview!
> 
> You wha?

Branch has always preview environment. ;) You dont have to screenshot.

**Comment by bryansamuel1** [comment]
> > > Please check always preview!
> > 
> > 
> > You wha?
> 
> Branch has always preview environment. ;) You dont have to screenshot.

how does one look at it?

**Comment by NikoGrano** [comment]
> > > > Please check always preview!
> > > 
> > > 
> > > You wha?
> > 
> > 
> > Branch has always preview environment. ;) You dont have to screenshot.
> 
> how does one look at it?

Gitlab => Operations => Environments

**Comment by NikoGrano** [comment]
🎉

**Pull request merged**